src/sim.rs

```diff
@@ -299,15 +301,24 @@ impl<'a> Sim<'a> {
 
         self.elapsed += tick;
 
-        // Check finished clients and hosts for err results. Runtimes are removed at
-        // this stage.
+        // Check finished clients and hosts for err results, ignoring cancelled
+        // tasks. Runtimes are removed at this stage.
         for addr in finished.into_iter() {
             if let Some(role) = self.rts.remove(&addr) {
                 let (rt, handle) = match role {
                     Role::Client { rt, handle } => (rt, handle),
                     Role::Simulated { rt, handle, .. } => (rt, handle),
                 };
-                rt.block_on(handle)??;
+                let result = rt.block_on(handle);
+                // Prevent JoinError::Cancelled to be propagated, assuming the task
+                // was crashed explicitly.
+                if let Err(e) = &result {
+                    if e.is_cancelled() {
+                        tracing::trace!(target: TRACING_TARGET, host = ?addr, "Crashed");
```

Comment:
Should we trace right when the crash happens (in bounce or crash) rather than when we await the handle?

Reply:
I was looking into that but confused how to log it. Let me try to get it right.